kore/test/Test/ConsistentKore.hs

```diff
@@ -610,41 +741,189 @@ maybeListBuiltinGenerator Setup { maybeListSorts } =
+        let concreteKeyGenerator :: Gen (Maybe (TermLike Concrete))
+            concreteKeyGenerator =
+                    requestNonSimplifiable
+                    $ concreteTermGenerator keySort childGenerator
+        maybeConcreteMap <-
+            Gen.map (Range.constant 0 5)
+                (   (,)
+                <$> concreteKeyGenerator
+                <*> childGenerator keySort
+                )
+        let concreteMapElem
+                :: (Maybe (TermLike Concrete), Maybe (TermLike Variable))
+                -> Maybe
+                    (TermLike Concrete, Domain.MapValue (TermLike Variable))
+            concreteMapElem (ma, mb) = (,) <$> ma <*> (Domain.MapValue <$> mb)
+            concreteMap =
+                Map.fromList
+                    (mapMaybe concreteMapElem (Map.toList maybeConcreteMap))
+        mixedKeyValues <-
+            requestNonSimplifiable
+            $ Gen.set (Range.constant 0 5)
+                (childGenerator keySort)
+        let variableKeys :: [TermLike Variable]
+            variableKeys =
+                filter
+                    (not . null . TermLike.freeVariables)
+                    (catMaybes (Set.toList mixedKeyValues))
+        maybeVariablePairs <- mapM variablePair variableKeys
+        let variablePairs :: [Domain.MapElement (TermLike Variable)]
+            variablePairs = catMaybes maybeVariablePairs
+        (Setup {metadataTools}, _) <- Reader.ask
+        return $ Just $
+            AssociativeCommutative.asInternalBuiltin
+                metadataTools
+                mapSort
+                (Domain.NormalizedMap Domain.NormalizedAc
+                    { elementsWithVariables = variablePairs
+                    , concreteElements = concreteMap
+                    , opaque = []
+                    -- TODO (virgil): also fill the opaque field.
+                    }
+                )
+      where
+        variablePair key = do
+            maybeValue <- childGenerator valueSort
+            return $ do  -- maybe monad
+                value <- maybeValue
+                return (Domain.MapElement (key, value))
+
+-- TODO(virgil): Test that we are generating non-empty sets.
 maybeSetBuiltinGenerator :: Setup -> Maybe BuiltinGenerator
 maybeSetBuiltinGenerator Setup { maybeSetSorts } =
     case maybeSetSorts of
         Nothing -> Nothing
-        Just _ -> error "Not implemented yet."
+        Just CollectionSorts {collectionSort, elementSort} ->
+            Just BuiltinGenerator
+                { arity = 10
+                , sort = SpecificSort collectionSort
+                , attributes = AttributeRequirements
+                    { isNonSimplifiable = False
+                    -- We could generate non-simplifiable or concrete sets
+                    -- if we wanted to.
+                    , isConcrete = False
+                    }
+                , generator = setGenerator collectionSort elementSort
+                }
+  where
+    setGenerator
```

Comment:
This seems similar to the `mapGenerator` above; are they similar enough that we could refactor?

Reply:
Done.

Reply:
Looks great, thanks!